src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -91,16 +109,48 @@ export default class ZipPaymentStrategy implements PaymentStrategy {
                             },
                         });
                     })
-                    .then(nonce =>
-                        this._store.dispatch(this._paymentActionCreator.submitPayment({
-                            methodId: payment.methodId,
-                            paymentData: { nonce },
-                        }))
+                    .then(nonce => {
+                            if (isNonceLike(nonce)) {
+                                return this._store.dispatch(this._paymentActionCreator.submitPayment({
+                                    methodId: payment.methodId,
+                                    paymentData: nonce,
+                                }));
+                            } else {
+                                this._goToOrderConfirmation();
+
+                                return this._store.getState();
+                            }
+                        }
                     ));
             });
     }
 
     finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
+
+    private _prepareForReferredRegistration(checkoutId: string): Promise<Response> {
+        const url = `checkout.php?action=zip_referred_registration`;
+        const options = {
+            headers: {
+                Accept: 'text/html',
+                'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+            },
+            body: toFormUrlEncoded({
+                checkoutId,
+            }),
+            method: 'post',
+        };
+
+        return this._requestSender.sendRequest(url, options);
+    }
+
+    private _goToOrderConfirmation() {
+        this._formPoster.postForm('/checkout/order-confirmation', {
```

Comment:
What are you trying to do here? Why are you doing HTML post to the order confirmation page? Do you just want to do a redirect? Also, we normally don't do any redirection to the order confirmation in the SDK because we let the client decide what they want to do immediately after a successful order submission. Having said that, we do have an exception for redirect payment methods where the callback happens directly from the provider / BigPay. But Zip seems to follow in-context flow instead. Therefore, resolving the promise seems to be a better solution, otherwise it'd be quite difficult to document.

Reply:
Yes, after the registration has entered the referred state Zip wants the shopper to be redirected to an order confirmation page ([docs](https://zip-online.api-docs.io/v1/checkout-and-registration/new-account-registration-flow-zip-money)) , how is the correct way of doing this?

Reply:
Yep, just resolve the promise, you don't need to do the redirect here. The checkout app will perform the redirection, just like all the other payment methods.